Ghidra/Features/Base/src/main/java/ghidra/program/flatapi/FlatProgramAPI.java

```diff
@@ -1514,6 +1514,35 @@ public final List<Symbol> getSymbols(String name, Namespace namespace) {
 	public final Namespace getNamespace(Namespace parent, String namespaceName) {
 		return currentProgram.getSymbolTable().getNamespace(namespaceName, parent);
 	}
+	
+	/**
+	 * Creates a new namespace with the given name contained inside the
+	 * specified parent namespace.
+	 * Pass <code>null</code> for namespace to indicate the global namespace.
+	 * If the namespace already exists then the existing one will be returned.
+	 * @param parent the parent namespace, or null for global namespace
+	 * @param namespaceName the requested namespace's name
+	 * @return the namespace with the given name
+	 * @throws InvalidInputException if the name is invalid
+	 */
+	public final Namespace createNamespace(Namespace parent, String namespaceName)
+			throws InvalidInputException {
+		SymbolTable symbolTable = currentProgram.getSymbolTable();
+		Namespace ns = symbolTable.getNamespace(namespaceName, parent);
+		if (ns != null) {
+			SymbolType type = ns.getSymbol().getSymbolType();
+			if (type == SymbolType.NAMESPACE || type == SymbolType.CLASS) {
```

Comment:
Javadoc should make it clear that an existing Namespace or GhidraClass could be returned.  Any thoughts on creation of GhidraClass?  You could link to SymbolTable.convertNamespaceToClass which may be useful.

Reply:
I didn't explicitly mention an existing `GhidraClass` since it is a `Namespace`. If you think it is clearer I will add it. I did want to add something to create a `GhidraClass` as well but wanted feedback first.

What would you think about a `createGhidraClass` method having the same behavior as `createNamespace` except it would allow a null name. If a null name is provided it would just convert the parent to a `GhidraClass`. I'm not sure if I should do that or just stick in a helper method to convert a namespace to a class as well.

Reply:
I've added a `createClass` method as well and included a link to `SymbolTable.convertNamespaceToClass` there as it seemed more fitting.